lib/course-db.js

```diff
@@ -188,9 +188,28 @@ module.exports.loadFullCourse = function(courseDir, logger, callback) {
         async.forEachOf(course.courseInstanceDB, function(courseInstance, courseInstanceDir, callback) {
             var assessmentsDir = path.join(course.courseInfo.courseInstancesDir, courseInstanceDir, 'assessments');
             courseInstance.assessmentDB = {};
-            that.loadInfoDB(courseInstance.assessmentDB, 'tid', assessmentsDir, 'infoAssessment.json', defaultAssessmentInfo, 'schemas/infoAssessment.json', null, null, course.courseInfo, logger, function(err) {
-                if (ERR(err, callback)) return;
-                callback(null);
+            // Check that the assessments folder exists and is accessible before loading from it
+            fs.access(assessmentsDir, function(err) {
+                if (err) {
+                    // If the directory does not exist, warn and skip assessment loading
+                    if (err.code == 'ENOENT') {
+                        logger.warn('Warning: ' + courseInstanceDir + ' has no `assessments` directory.');
+                    
+                    }
+                    // Report the error code for any other error and skip assessment loading
+                    else {
+                        logger.warn('Warning: ' + courseInstanceDir + ' `assessment` directory inaccessible (lstat error code: ' + err.code + ').');
```

Comment:
nit: `assessment` -> `assessments`. Also a good spot for template strings!

Reply:
Agreed!

Reply:
I've not used template strings before -- I'll take a look later today.